### ISSUE TITLE
ADFS Authentication fails while using TestCafe version 1.14.1+ (1.15.0/1.16.0)

### ISSUE THREAD

**Issue by aarti315** [DevExpress/testcafe#6566]
[TestCafeJSScript.txt](https://github.com/DevExpress/testcafe/files/7225112/TestCafeJSScript.txt)
We were using TestCafe 1.10.1 where scripts used to work fine. Scripts also works fine with 1.14.0. If we upgrade to latest version of TestCafe then TestCafe scripts stops working as it gets stuck at authentication step.

### What is your web application and your TestCafe test code?

 
<details>
<summary>Your website URL (or attach your complete example):</summary>

</details>

<details>
<summary>Your complete test code (or attach your test files):</summary>

 
```js
 
```
</details>

<details>
<summary>Your complete configuration file (if any):</summary>

```

```
</details>

<details>
<summary>Your complete test report:</summary>

```

```
</details>

<details>
<summary>Screenshots:</summary>

```

```
</details>

### Steps to Reproduce:

1. Execute attach script using version 1.10.1/1.14.1 (where it works fine)
2. Execute attach script using version 1.15.0/1.16.0.1 (where it stopped working)

### Your Environment details:
 
* testcafe version:                   
* node.js version:                    
* command-line arguments:  
* browser name and version: 
* platform and version:          
* other:

**Comment by aarti315** [comment]
You can also try reproducing the issue with the attached script as earlier one had an
[TestCafeJSScript_6566.txt](https://github.com/DevExpress/testcafe/files/7225241/TestCafeJSScript_6566.txt)
 issue.

**Comment by pavsenin** [comment]
Hello aarti315,
Thank you for the code samples.
I have tried to reproduce the issue but without success.
I have exactly the same result with the following testcafe versions: 1.14.0, 1.15.0, 1.16.0.

The fixture from the TestCafeJSScript.txt file has the following output:
```
Using locally installed version of TestCafe.
 Running tests in:
 - Chrome 93.0.4577.82 / Windows 10

 Link Verification
 √ Usermenu Dropdown - Security Overview
undefined
undefined
ERROR The Chrome 93.0.4577.82 / Windows 10 browser disconnected. If you did not close the browser yourself, browser performance or network issues may be at fault.

Type "testcafe -h" for help.
undefined
```
The fixture from the TestCafeJSScript_6566.txt file has the following output:
```
Using locally installed version of TestCafe.
 Running tests in:
 - Chrome 93.0.4577.82 / Windows 10

 Link Verification
 √ Usermenu Dropdown - Security Overview

 1 passed (40s)
```
 
So, since I have the same results with different versions of testcafe, it seems like the issue may be related to some specifics of your environment. Can you try to reproduce the issue in another environment/on another machine?

**Comment by aarti315** [comment]
Thanks for your response. I will check again on my side and come back in case of any issues.

**Comment by aarti315** [comment]
I check and found the issue is only observed in case of internal user and not for external/third party user. There is difference in authentication method user for different users. For internal user, ADFS is used and we also use SSO. I need to investigate more on authentication details to find the difference.

Just in case if there is any such issue reported earlier then let me knw.

**Comment by pavsenin** [comment]
Hello,

Thank you for the clarification.

We have a [similar issue](https://github.com/DevExpress/testcafe/issues/6537) with ADFS \ SSO authentication, but we did not manage to get the required information to reproduce and further research it from the customer.

Please share the results of your research with us.

**Comment by omnidevZero** [comment]
I ran into the same issue. ADFS redirection ends up with a GET request status 400 and a blank page. Works fine with 1.14.0 but updating to 1.14.1 or anything above that causes the issue to occur. I can't provide you with an environment to reproduce it so I'm in just to let you know this might be a widespread problem regarding ADFS redirection during authentication.

**Comment by pavsenin** [comment]
omnidevZero,

Thank you for letting us know the results of your research.
The issue with the --disable-http2 CLI option is already fixed. The fix is available in testcafe@1.16.1-alpha.1.

Regarding, the initial issue with the ADFS, it would be helpful if we could get an environment where this issue is reproducible for further research.
